xdsl/irdl/declarative_assembly_format_parser.py

```diff
@@ -124,30 +127,34 @@ def parse_format(self) -> FormatProgram:
         elements: list[FormatDirective] = []
         while self._current_token.kind != Token.Kind.EOF:
             elements.append(self.parse_directive())
-            last = elements[-1]
-            if (
-                len(elements) >= 2
-                and isinstance(last, PunctuationDirective)
-                and last.punctuation == ","
-                and isinstance(
-                    elements[-2],
-                    VariadicOperandTypeDirective
-                    | VariadicResultTypeDirective
-                    | VariadicOperandVariable,
-                )
-            ):
-                self.raise_error(
-                    "A variadic directive cannot be followed by a comma literal."
-                )
 
         self.add_reserved_attrs_to_directive(elements)
         seen_variables = self.resolve_types()
+        self.verify_directives(elements)
         self.verify_attr_dict()
         self.verify_properties()
         self.verify_operands(seen_variables)
         self.verify_results(seen_variables)
         return FormatProgram(elements)
 
+    def verify_directives(self, elements: list[FormatDirective]):
```

Comment:
```suggestion
    def verify_variadic_directives(self, elements: list[FormatDirective]):
```
To make this more obvious?
I would also add documentation on this function